### PR TITLE
kie-issues#1613: Add `CI :: Check license headers` GitHub Actions workflow

### PR DESCRIPTION
**Pull request by jomarko**
This is partial fix for https://github.com/apache/incubator-kie-issues/issues/1613. We open one PR per repository.

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Release notes updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] tests</b>
  
- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: 
    please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: 
    add the label `run_fdb`

- for a <b>compile downstream build</b>  
  please add comment: <b>Jenkins run cdb</b>

- for a <b>full production downstream build</b>  
  please add comment: <b>Jenkins execute product fdb</b>

- for an <b>upstream build</b>  
  please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] quarkus-branch</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] native</b>

- for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] native-lts</b>

</details>

### CI Status

 You can check OptaPlanner repositories CI status from [Chain Status webpage](https://apache.github.io/incubator-kie-optaplanner/).

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by jomarko** [comment]
This will be addressed via https://github.com/apache/incubator-kie-issues/issues/1613

**Review by yesamer — APPROVED** [review]

File: NOTICE [hunk 1]
```diff
@@ -10,22 +10,10 @@ Copyright Red Hat, Inc. and/or its affiliates.. All Rights Reserved.
...
   Downloaded from: http://code.google.com/p/google-code-prettify
   License: Apache License v2.0 (just like OptaPlanner)
 
-* jQuery version: 3.3.1
-  Downloaded from: http://jquery.com/
-  License: MIT - this is compatible with ASL 2.0: http://www.apache.org/legal/resolved.html#category-a
-
 * jstree version: 3.3.5
   Downloaded from: https://www.jstree.com/
   License: MIT - this is compatible with ASL 2.0: http://www.apache.org/legal/resolved.html#category-a
```
Comment (tkobayas) [inline NOTICE:19]: Thank you for the great work, @jomarko !

As mentioned in https://github.com/apache/incubator-kie-issues/issues/1616#issue-2656082830 , `NOTICE` file only contains contents of `NOTICE` files of the third-party components.

I confirmed that the repositories below don't have `NOTICE` files, so we can simply remove these items from our `NOTICE` file.

- https://github.com/vakata/jstree/
- https://github.com/googlearchive/code-prettify
- https://github.com/twbs/bootstrap/
- https://github.com/jquery/jquery

File: LICENSE.txt [hunk 2]
```diff
@@ -200,3 +200,62 @@
...
+    LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+    OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+    WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+--------------------------------------------------------------------------------------------------------------------
+for optaplanner-benchmark/src/main/resources/org/optaplanner/benchmark/impl/report/twitterbootstrap/css/prettify.css
+    optaplanner-benchmark/src/main/resources/org/optaplanner/benchmark/impl/report/twitterbootstrap/js/prettify.js
+
+    (https://github.com/googlearchive/code-prettify)
+    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (tkobayas) [inline LICENSE.txt:261]: The file name `LICENSE.txt` is permitted, but it's better to align with `LICENSE` across all repos. (CI will likely need to combine the files for the root directory)

File: NOTICE (hunk 1 above)
Comment (tkobayas) [reply NOTICE:19]: Also please add jstree to `LICENSE` file. (If `optaplanner-docs` is excluded from source distribution using the release CI, we don't even need to add this to `LICENSE`)

```
optaplanner-docs/src/modules/ROOT/images/website/jstree/32px.png
optaplanner-docs/src/modules/ROOT/images/website/jstree/40px.png
optaplanner-docs/src/modules/ROOT/images/website/jstree/jstree.js
optaplanner-docs/src/modules/ROOT/images/website/jstree/jstree.min.js
optaplanner-docs/src/modules/ROOT/images/website/jstree/style.css
optaplanner-docs/src/modules/ROOT/images/website/jstree/throbber.gif
```

File: LICENSE.txt (hunk 2 above)
Comment (jomarko) [reply LICENSE.txt:261]: renamed https://github.com/apache/incubator-kie-optaplanner/pull/3130/commits/5b40428765f59b1f62d169d1b7e4c35d48b5e0f0

File: NOTICE (hunk 1 above)
Comment (jomarko) [reply NOTICE:19]: `optaplanner-docs` is included in the source distribution. I will add corresponding entry into `LICENSE` for the `jstree`

**Comment by jomarko** [comment]
@tkobayas thank you for the help, changes pushed

File: NOTICE [hunk 3]
```diff
@@ -1,31 +0,0 @@
-Apache KIE
-Copyright 2024 The Apache Software Foundation
-
-This product includes software developed at
-The Apache Software Foundation (http://www.apache.org/).
-
-The Initial Developer of some parts of the framework, which are copied from, derived from, or
-inspired by KIE (Knowledge Is Everthing) group, is Red Hat, Inc (https://www.redhat.com/).
-Copyright Red Hat, Inc. and/or its affiliates.. All Rights Reserved.
```
Comment (tkobayas) [inline NOTICE:9]: @jomarko Ahh, I'm sorry that I confused you. I wanted to say we can remove the items for third party components. But we need to keep the main notice about Apache KIE. So the result would be like the same as https://github.com/apache/incubator-kie-drools/blob/main/NOTICE , not removing the whole file.

Comment (jomarko) [reply NOTICE]: @tkobayas sorry, my fault, I added the NOTICE file back

**Review by porcelli — APPROVED** [review]

**Pull request merged**